### PR TITLE
MuonAlignment GEM phase2 geometry

### PR DESCRIPTION
**Pull request by hyunyong**
#### PR description:
Run3 geometry has a GEM station 2 demostration chamber, so we need to remove unessesary if statement for the AlignableMuon::buildGEMEndcap function. 

#### PR validation:
Test with current GT and GEM geomtry.

### CODE REVIEW

**Comment by tvami** [comment]
Hi @hyunyong I believe this change should guarded by the `run3_common` modifier. This current approach would make the code break down for Run-2 and Run-1 data.

**Comment by hyunyong** [comment]
> Hi @hyunyong I believe this change should guarded by the `run3_common` modifier. This current approach would make the code break down for Run-2 and Run-1 data.

Hi @tvami, thank you for your comment.
Yes, currently, the alignbleMuon class is not considered the Run 1 or 2 data.
The alignbleMuon class takes geometries so it should be fine with any GEM geometry but if there is no GEM geometry it will be a problem.
I will try to fix this problem but it is complicated.
Also, it is not propagated to any other CMSSW modules.
It is related to only the muon alignment.
However, we just need to fix this to upload the GEM alignment tag.

**Comment by tvami** [comment]
> Yes, currently, the alignbleMuon class is not considered the Run 1 or 2 data.

I understand, but the CMSSW has to be ready for that too. This is why we have era modifiers.

**Comment by hyunyong** [comment]
> Hi @hyunyong any updates on the Run-3 modifier?

I'm thinking about this but it seems we need to re-write the entire muon alignment modules. 
The muonAlignment class constructor takes three muon subsystem geometries, so we need to change it.
However, it propagates to a lot of muon alignment codes.
If you have any suggestions, please let me know.

File: Alignment/MuonAlignment/src/AlignableMuon.cc [hunk 1]
```diff
@@ -287,10 +287,6 @@ void AlignableMuon::buildGEMEndcap(const GEMGeometry* pGEM, bool update) {
     std::vector<AlignableGEMRing*> tmpGEMRingsInStation;
 
     for (int ist = 0; ist < 3; ist++) {
-      if (ist == 0)
-        continue;  //Run3 GEM dosen't have ME0
-      if (ist == 2)
-        continue;  //Run3 GEM dosen't have GE2/1
```
[the remark this answers is not in the document]
Comment (tvami) [reply Alignment/MuonAlignment/src/AlignableMuon.cc:293]: That's the same to what I suggested, no?

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca
 * tests pass
 * I dont think there is a unit test for Muon alignment, it would be nice to add that @hyunyong

**Comment by qliphy** [comment]
+1

**Comment by hyunyong** [comment]
> +alca
> 
> * tests pass
> * I dont think there is a unit test for Muon alignment, it would be nice to add that @hyunyong

Yes, thank you!

**Pull request merged**